sdk/resourcemanager/Azure.ResourceManager.Core/src/Generated/TagsOperations.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary> The Tags service client. </summary>
+    public class TagsOperations : OperationsBase
+    {
+        /// <summary>
+        /// The resource type for Tags
+        /// </summary>
+        public static readonly ResourceType ResourceType = "Microsoft.Resources/tags";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TagsOperations"/> class for mocking.
+        /// </summary>
+        protected TagsOperations()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TagsOperations"/> class.
+        /// </summary>
+        /// <param name="clientContext"></param>
+        /// <param name="subscriptionGuid"> The Guid of the subscription. </param>
+        internal TagsOperations(ClientContext clientContext, string subscriptionGuid)
+            : base(clientContext, new SubscriptionResourceIdentifier(subscriptionGuid))
+        {
+        }
+
+        /// <summary>
+        /// Gets the valid resource type for this operation class.
+        /// </summary>
+        protected override ResourceType ValidResourceType => ResourceType;
+
+        private TagsRestOperations RestClient 
+        {
+            get
+            {
+                string subscriptionId;
+                if (Id.TryGetSubscriptionId(out subscriptionId))
```

Comment:
we know the Id will be of type SubscriptionResourceIdentifier.  Lets make a private variable of this type and then we can simply call Id.SubscriptionId

Reply:
Done